### ISSUE TITLE
Feat: Support direct import of cluster certificates from other platforms (such as ACK, EKS)

### ISSUE THREAD

**Issue by elliotxx** [KusionStack/karpor#458]
### What would you like to be added?

Support direct import of cluster certificates from other platforms (such as ACK, EKS)

### Why is this needed?

Currently, only manual upload of cluster certificates is supported, which is not simple and direct for users using the cloud platform.

**Comment by ruquanzhao** [comment]
**Proposal: Cloud Credential Management**

Add a new GVK called `cloudcredentials` to store users' cloud credentials, and then reference these credentials in `cluster.karpor.io/v1beta1/cluster`.

Below is the user workflow:

**Dashboard:**
1. Create a cloudCredential in the dashboard.
2. Upload a kubeconfig, and select to connect to a specific cloudCredential. By doing this, the user field in the kubeconfig will be ignored, and the client will be built using this cloudCredential.

**CLI:**

1. Create a cloudCredential using a YAML file.
2. Create a cluster that references the cloudCredential using a YAML file.

**Comment by ruquanzhao** [comment]
cc @elliotxx @fanfan-yu

**Comment by fanfan-yu** [comment]
Hi. I'm interested in what the advantage of using a new `crd`. If this solution is more secure, it will bring new changes: how to storage this crds.

**Comment by elliotxx** [comment]
@ruquanzhao Cool, let me see the details 👍

**Comment by ruquanzhao** [comment]
> Hi. I'm interested in what the advantage of using a new `crd`. If this solution is more secure, it will bring new changes: how to storage this crds.

Using the same CRD allows users to more easily maintain clusters under the same cloud account.

> it will bring new changes: how to storage this crds.

@fanfan-yu You are right. Security is another important point. Currently, anyone with the Karpor admin certificate can access the credentials for all clusters. Ideally, these `credentials` and `crd cloud-credentials` should be encrypted and only decrypted at runtime.

**Comment by jinjiaKarl** [comment]
Managing multi-cloud Kubernetes clusters seems to be a common challenge, so I'm curious—do you know how other Kubernetes visualization projects handle this?

**Comment by elliotxx** [comment]
Yes, I am investigating similar projects, such as Lens, Karmada, etc., and there are no specific results yet; but I think everyone's approach is similar, and it takes time to fill in this part of the functions, and there may be other specific issues created.
A little busy on the weekend, I reply a little late,sorry

**Comment by jinjiaKarl** [comment]
Cool, Appreciate it, I'm willing to see how this goes and make some contributions if needed.

**Comment by elliotxx** [comment]
Great brother! If I have formed content later, I will create an issue, which should be some specific work cc @fanfan-yu can also take a look together

**Comment by CirillaQL** [comment]
Hi, I am the guy who supports the AWS kubeconfig auth method. Recently, I checked other projects which are trying to connect to multiple clusters. Some of them are using agents to transform data. They deploy a deployment into each cluster. Have you guys thought about this approach?

**Comment by CirillaQL** [comment]
@elliotxx There is a Kubernetes workload AI controller system called KubeDoor. I believe they are using an agent to accomplish that. Also, as far as I know, Karmada also uses an agent to connect to the control plane. 
I understand that it is a big change for Karpor. I also read the CRD method you mentioned at the top. If you guys choose a solution to implement it, please let me know.

**Comment by fanfan-yu** [comment]
Good idea! We will implement agent the in high-availability mode, which like the solution in `karmada` (details is in `https://github.com/KusionStack/karpor/tree/feat-karpor-agent`). However, it will bring complexity for users, so we will discuss with more comprehensive solution. 

If you have better suggestions, welcome to discuss.

**Comment by CirillaQL** [comment]
@fanfan-yu Thank you! I didn't notice that feature. I will read through the code and discuss it later.
